src/TLuaInterpreterMudletObjects.cpp

```diff
@@ -2555,3 +2556,114 @@ int TLuaInterpreter::tempTrigger(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getProfiles
+int TLuaInterpreter::getProfiles(lua_State* L)
+{
+    auto& hostManager = mudlet::self()->getHostManager();
+    const QStringList profiles = QDir(mudlet::getMudletPath(mudlet::profilesPath))
+                                   .entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+
+    lua_newtable(L);
+
+    for (const auto& profile : profiles) {
+        lua_pushstring(L, profile.toUtf8().constData());
+        lua_newtable(L);
+
+        QString url = mudlet::self()->readProfileData(profile, qsl("url"));
+        QString port = mudlet::self()->readProfileData(profile, qsl("port"));
+
+        // if url/port haven't been written to disk yet (which is what happens
+        // when a default profile is opened for the first time), fetch this data from game details
+        if (url.isEmpty()) {
+            auto it = TGameDetails::findGame(profile);
+            if (it != TGameDetails::scmDefaultGames.end()) {
+                url = (*it).hostUrl;
+            }
+        }
+        if (port.isEmpty()) {
+            auto it = TGameDetails::findGame(profile);
+            if (it != TGameDetails::scmDefaultGames.end()) {
+                port = QString::number((*it).port);
+            }
+        }
+
+        if (!url.isEmpty()) {
+            lua_pushstring(L, "host");
+            lua_pushstring(L, url.toUtf8().constData());
+            lua_settable(L, -3);
+        }
+
+        if (!port.isEmpty()) {
+            lua_pushstring(L, "port");
+            lua_pushstring(L, port.toUtf8().constData());
+            lua_settable(L, -3);
+        }
+
+        auto host = hostManager.getHost(profile);
+        const bool loaded = host != nullptr;
+        lua_pushstring(L, "loaded");
+        lua_pushboolean(L, loaded);
+        lua_settable(L, -3);
+
+        if (loaded) {
+            auto [hostName, hostPort, connected] = host->mTelnet.getConnectionInfo();
+
+            lua_pushstring(L, "connected");
+            lua_pushboolean(L, connected);
+            lua_settable(L, -3);
+        }
+
+        lua_settable(L, -3);
+    }
+
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#loadProfile
+int TLuaInterpreter::loadProfile(lua_State* L)
+{
+    auto& hostManager = mudlet::self()->getHostManager();
+    const QString profileName = getVerifiedString(L, __func__, 1, "profile name");
+    bool offline = false;
+
+    if (lua_gettop(L) > 1) {
+        if (!lua_isboolean(L, 2)) {
+            lua_pushfstring(L, "loadProfile: bad argument #2 type (offline mode as boolean expected, got %s!)", luaL_typename(L, 2));
+            return lua_error(L);
+        }
+        offline = lua_toboolean(L, 2);
```

Comment:
It is "optional" not "expected" - and ***we have a function for either case for this***:
```suggestion
        offline = getVerifiedBool(L, __func__, 2, "offline mode", true);
```
The only reason perhaps not to use it is if we wanted to expand the text to mention that it defaults to `false` if not provided - as we do not have that ability in the current version of the `getVerifiedXxxx(...)` functions...